src/components/badges/BadgeTitle.tsx

```diff
@@ -0,0 +1,36 @@
+import BaseProof from 'helpers/BaseProof'
+import ContractName from 'components/ContractName'
+import ERC721Proof from 'helpers/ERC721Proof'
+import EmailProof from 'helpers/EmailProof'
+import ExternalLink from 'components/ExternalLink'
+import getEtherscanAddressUrl from 'helpers/getEtherscanAddressUrl'
+
+function ProofName({ badge }: { badge: BaseProof }) {
+  if (badge instanceof ERC721Proof)
+    return <ContractName address={badge.contract} />
+  if (badge instanceof EmailProof) return <>{badge.domain}</>
+  return <>Unknown</>
+}
+
+export default function ({
+  derivativeAddress,
+  proof,
+}: {
+  derivativeAddress?: string
+  proof?: BaseProof
+}) {
+  if (derivativeAddress)
+    return (
+      <ExternalLink url={getEtherscanAddressUrl(derivativeAddress)}>
+        <ContractName address={derivativeAddress} />
+      </ExternalLink>
+    )
+
+  if (!proof) return null
+
+  return (
+    <>
+      <ProofName badge={proof} /> (derivative)
```

Comment:
We also have `(email)` text in designs
![image](https://user-images.githubusercontent.com/49658988/175802078-a7d87935-ab8e-45c0-8b59-e5cfb90ffcee.png)

Reply:
Also, what will this function return if you pass `derivativeAddress`, but won't pass `proof`?

Reply:
For simplicity added just return of the name of the derivative contract